src/proptests.rs

```diff
@@ -0,0 +1,61 @@
+use proptest::test_runner::{TestRunner};
+use proptest::strategy::{Strategy, ValueTree};
+use proptest::prelude::*;
+
+proptest! {
+    #[test]
+    fn test_fixed_a(_ in 0..1_000_000_000) {
+        let minimal_failing_case = find_minimal_failing_case();
+
+        if minimal_failing_case.is_some() {
+            let val = minimal_failing_case.unwrap();
+
+            assert_eq!(val.0, 1);
+            assert!(val.1 >= 33499370958229846944519890329592825);
+            assert!(val.2 >= 33499370958229846944519890329592825);
+        }
+    }
+}
+
+
+/// Trying to find minimal failing cases for #calculate_spot_price
+#[allow(unused)]
```

Comment:
When trying to find minimal failing cases, we can make the variable sell_reserve constant and try to simplify and complicate buy_amount & buy_reserve, as the formula is:

/// Formula : BUY_RESERVE * AMOUNT / SELL_RESERVE

Hence the upper bound values for buy_reserve & amount are determined by sell_reserve.

It is quite tricky to implement Strategies for this one without duplicating the function logic.
Of course I am not satisfied with this, however we could at least make (almost) guarantee a correct result if buy_reserve and amount each are lower than 33499370958229846944519890329592825.